File: frontend/src/app/service/attendance.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { BASE_URL } from "../constant/base.url";
import { AttendanceInsertReqDto } from "../dto/attendance/attendance.insert.req.dto";
import { AttendanceListResDto } from "../dto/attendance/attendance.list.res.dto";
import { AttendanceUpdateReqDto } from "../dto/attendance/attendance.update.req.dto";

@Injectable({
    providedIn: 'root'
})
export class AttendanceService {
    constructor(private http: HttpClient) { }
    
    insert(data :AttendanceInsertReqDto) : Observable<AttendanceInsertReqDto>{
        return this.http.post<AttendanceInsertReqDto>(`${BASE_URL.LOCALHOST}/attendance`, data);
    }
    
    getAttendExam(id:number) : Observable<AttendanceListResDto>{
        return this.http.get<AttendanceListResDto>(`${BASE_URL.LOCALHOST}/attendance/exam_schedule/${id}`);
    }

    getAttendQuiz(id:number) : Observable<AttendanceListResDto>{
        return this.http.get<AttendanceListResDto>(`${BASE_URL.LOCALHOST}/attendance/quiz_schedule/${id}`);
    }

    getAttendLearning(id:number) : Observable<AttendanceListResDto>{
        return this.http.get<AttendanceListResDto>(`${BASE_URL.LOCALHOST}/attendance/learning_schedule/${id}`);
    }

    update(data :AttendanceUpdateReqDto) : Observable<AttendanceUpdateReqDto>{
        return this.http.put<AttendanceUpdateReqDto>(`${BASE_URL.LOCALHOST}/attendance`, data);
    }
}

File: src/main/java/com/lawencon/elearn/service/AttendanceService.java
package com.lawencon.elearn.service;

import com.lawencon.elearn.dto.attendance.AttendanceInsertReqDto;
import com.lawencon.elearn.dto.attendance.AttendanceInsertResDto;
import com.lawencon.elearn.dto.attendance.AttendanceListResDto;
import com.lawencon.elearn.dto.attendance.AttendanceResDto;
import com.lawencon.elearn.dto.attendance.AttendanceUpdateReqDto;
import com.lawencon.elearn.dto.attendance.AttendanceUpdateResDto;

public interface AttendanceService {
	AttendanceListResDto getAll();
	AttendanceResDto getById(Long id);
	AttendanceInsertResDto insert(AttendanceInsertReqDto data);
	AttendanceUpdateResDto update(AttendanceUpdateReqDto data);
	AttendanceListResDto getAttendLearn(Long scheduleId);
	AttendanceListResDto getAttendQuiz(Long scheduleId);
	AttendanceListResDto getAttendExam(Long scheduleId);
//	AttendanceUpdateResDto approvLearn(AttendanceUpdateResDto data);
//	AttendanceInsertResDto attend(AttendanceInsertResDto data);
}
